vividus-plugin-web-app/src/main/java/org/vividus/selenium/WebDriverFactory.java

```diff
@@ -25,36 +29,48 @@
 
 import org.apache.commons.lang3.StringUtils;
 import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.HasAuthentication;
+import org.openqa.selenium.UsernameAndPassword;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
 import org.openqa.selenium.remote.CapabilityType;
 import org.openqa.selenium.remote.DesiredCapabilities;
 import org.openqa.selenium.support.events.EventFiringDecorator;
 import org.openqa.selenium.support.events.WebDriverListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.vividus.proxy.IProxy;
+import org.vividus.selenium.authentication.BasicAuthCredentials;
+import org.vividus.selenium.driver.DelegatingWebDriver;
 import org.vividus.selenium.manager.WebDriverManager;
 import org.vividus.ui.web.listener.WebDriverListenerFactory;
 import org.vividus.util.json.JsonUtils;
+import org.vividus.util.property.IPropertyMapper;
 import org.vividus.util.property.IPropertyParser;
 
 public class WebDriverFactory extends GenericWebDriverFactory
 {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WebDriverFactory.class);
+
     private static final String COMMAND_LINE_ARGUMENTS = "command-line-arguments";
+    private static final String BASIC_AUTH_PREFIX = "basic-authentication-";
 
     private final IProxy proxy;
     private final WebDriverStartContext webDriverStartContext;
     private final TimeoutConfigurer timeoutConfigurer;
+    private final IPropertyMapper propertyMapper;
     private WebDriverType webDriverType;
     private final List<WebDriverListenerFactory> webDriverListenerFactories;
     private final boolean remoteExecution;
 
     private final Map<WebDriverType, WebDriverConfiguration> configurations = new ConcurrentHashMap<>();
 
+    @SuppressWarnings("paramNum")
     public WebDriverFactory(boolean remoteExecution, IRemoteWebDriverFactory remoteWebDriverFactory,
             IPropertyParser propertyParser, JsonUtils jsonUtils, IProxy proxy,
             WebDriverStartContext webDriverStartContext,
             Optional<Set<DesiredCapabilitiesAdjuster>> desiredCapabilitiesAdjusters,
-            TimeoutConfigurer timeoutConfigurer,
+            TimeoutConfigurer timeoutConfigurer, IPropertyMapper propertyMapper,
```

Comment:
it's better to inject here `PropertyMappedCollection<BasicAuthCredentials>`:
- all logic of working with properties is hidden and separated from the code
- no need to read it everytime we instantiate WbeDriver